python-package/lightgbm/callback.py

```diff
@@ -254,10 +257,10 @@ def __init__(
         self._reset_storages()
 
     def _reset_storages(self) -> None:
-        self.best_score = []
-        self.best_iter = []
-        self.best_score_list = []
-        self.cmp_op = []
+        self.best_score: List[float] = []
+        self.best_iter: List[int] = []
+        self.best_score_list: List[_EvalResultTuple] = []
```

Comment:
```suggestion
        self.best_score_list: List[Union[_EvalResultTuple, None]] = []
```

Please include the `None` here in this hint, instead of as a part of `_EvalResultTuple`

Reply:
Done :)